src/pages/manage_dac/EditDac.jsx

```diff
@@ -83,44 +82,42 @@ export default function EditDac(props) {
 
   const okHandler = async (event) => {
     event.preventDefault();
-
+    const user = Storage.getCurrentUser();
     let currentDac = state.dac;
     if (state.dirtyFlag) {
-      if (props.location.state.userRole === ADMIN) {
-        if (dacId !== undefined) {
-          await DAC.update(currentDac.dacId, currentDac.name, currentDac.description, currentDac.email);
+      if (dacId !== undefined) {
+        await DAC.update(currentDac.dacId, currentDac.name, currentDac.description, currentDac.email);
+      } else {
+        if (daaFileData === null && selectedDaa.daaId !== broadDaa.daaId) {
+          handleErrors('Please select either the default agreement or upload your own agreement before saving.');
+          return;
+        } else if (user.isAdmin && daaFileData !== null && selectedDaa === undefined) {
+          currentDac = await DAC.create(currentDac.name, currentDac.description, currentDac.email);
+          const createdDaa = await DAA.createDaa(daaFileData, currentDac.dacId);
+          setCreatedDaa(createdDaa.data);
         } else {
-          if (daaFileData === null && selectedDaa.daaId !== broadDaa.daaId) {
-            handleErrors('Please select either the default agreement or upload your own agreement before saving.');
-            return;
-          } else if (daaFileData !== null && selectedDaa === undefined) {
-            currentDac = await DAC.create(currentDac.name, currentDac.description, currentDac.email);
-            const createdDaa = await DAA.createDaa(daaFileData, currentDac.dacId);
-            setCreatedDaa(createdDaa.data);
-          } else {
+          if (user.isAdmin) {
             currentDac = await DAC.create(currentDac.name, currentDac.description, currentDac.email);
```

Comment:
I'm having trouble following the different cases here. If there is a dacId, update the DAC, else if there is non broad DAA selected without a file throw an error, else if the user is an admin and there is DAA file data, create a DAC and a DAA for it, else if the user is an admin create a dac. This is pretty complex and not super clear, I think it could benefit from either comments or helper functions or something to make this flow easier to return to in the future.

Reply:
I'm trying to maintain the existing logic minus the chairperson restriction that existed before. I can take another stab at trying to refactor this, but my primary goal is bug fixing.

Reply:
I don't want to prevent progress/bug fixing here if this is too big of a lift, but component tests for this sort of complex logic might be helpful to confirm it is working as expected.

Reply:
Good idea ... let me see what I can do here.

Reply:
@raejohanek - Another way to look at this is to hide the white space in the PR. That should make the actual changes much easier to see.

![Screenshot 2024-10-31 at 1 03 06 PM](https://github.com/user-attachments/assets/2e649275-4318-4f34-9764-aea7329dcc60)